frontend/src/metabase-types/api/collection.ts

```diff
@@ -5,7 +5,7 @@ import type { DatabaseId } from "./database";
 
 export type RegularCollectionId = number;
 
-export type CollectionId = RegularCollectionId | "root" | "personal";
```

Comment:
The reason `personal` is here is because we're using it in FE. This ID won't be from BE.

https://github.com/metabase/metabase/blob/df746e7015b0d41444ff53fd5c6653d0c870497c/frontend/src/metabase/entities/collections/constants.ts#L24

![Screenshot 2023-12-13 at 2 40 01 PM](https://github.com/metabase/metabase/assets/1937582/6135a581-ce90-4535-bb40-88f86b7d9eb4)

I think we should still leave it here.

Reply:
The comment above it says:
`fake collection for admins that contains all other user's collections`

We already have a string id for that - it's called `users` and it's working already with a router. Is there a reason why we are not using that string?

At the very least, if we insist on using `personal` instead of `users` for ItemPicker, I should go and expand the type because TS will scream at you in certain cases. `users` is a valid collection id that's missing from this type.

Reply:
I meant that we should not remove `personal`, but I'm fine with adding `users`. Hope that's clear.

Reply:
That is clear.
I was just wondering what's the technical reason for choosing `personal` instead of the existing `users`? Maybe @ranquild remembers?

Reply:
Ahh I got it now. I'm sorry that I missed your point earlier. I think we should be able to replace `personal` with `users`. But I don't know the details. I just know we have `personal` because I've worked on an area that uses `personal` as a collection name.

Reply:
Thank you.

Yeah, I don't mind updating all those references if @ranquild doesn't object to using `users` instead of `personal`. IMO, that approach is more clear and it removes an implicit the implicit abstraction.

OTOH, if we decide to leave `personal` along side `users` we ought to document it using JSDoc directly on the type. But I'm hoping to replace it completely with the existing type to avoid all this unnecessary hacking.

Reply:
I don't think that "users" is a collection ID unless it's supported on the backend. That fact we use this string for routing doesn't mean it's a collection id

Reply:
@ranquild sorry, but that doesn't answer the question, not does it give any pointers as to how to proceed.

Even if both `personal` and `users` are just FE string IDs that backend doesn't recognize, my question still stands. Why use `users` in router and then invent `personal` for other other UI-related cases?

I'm trying to understand the reasoning behind that decision because it's not documented anywhere. If there were some technical factors that affected that decision, that's helpful to know. I cannot arrive to the good solution without knowing these details.

To re-iterate:
Was there a technical reason we chose `personal` as ID identifier?
    a. If not, do you feel strongly about replacing it with `users` that we already use for routing?
    b. If yes, can you please briefly explain it, and I'll just add another string to the type, without removing anything.
    
Thank you

Reply:
Whatever string we use for routing doesn't mean we should patch our API types. I'd not do this change.